gigadb/app/tools/transfer-files/README.md

```diff
@@ -0,0 +1,513 @@
+scripts/transfer.sh --doi 100148 --sourcePath /share/dropbox/user101 --wasabi --backup --apply
+
+```
+
+User has to specify the `--doi` which defines the destination folder name, and the `--sourcePath` defines which file or directory to be copied over.
+
+`transfer.sh` is a script that can execute 2 rclone copy actions:
+
+1. `--wasabi` option will copy files from local to wasabi bucket
+2. `--backup` option will copy files from local s3 bucket in glacier tier
+3. These two options can be supplied separately or together 
+
+`transfer.sh` would execute the copy actions in dry-run mode by default, unless option `--apply` is supplied.
+
+And the log output of the `transfer.sh` would be saved to `/gigadb/app/tools/transfer-files/log` dir or in `/var/log/gigadb/` in production environments.
+The format of the log output file name is in `transfer.log`
+
+Define the profile name [aws-transfer] and [wasabi-transfer] in the /home/$user/~.aws/credentials, 
+by adding parameter [s3-profile](https://rclone.org/s3/#s3-profile) to the rclone cmd, the rclone will then able to get the corresponding secrets from /home/$user/~.aws/credentials
+
+### Pre-requisite
+1. rclone is installed in dev, eg:
+```
+ % rclone --version
+rclone v1.67.0
+- os/version: darwin 13.4.1 (64 bit)
+- os/kernel: 22.5.0 (arm64)
+- os/type: darwin
+- os/arch: arm64 (ARMv8 compatible)
+- go/version: go1.22.4
+- go/linking: dynamic
+- go/tags: none
+```
+2. The following variables need to be added into your gitlab project variable page
+
+| key | value | env     |
+| --- | --- |---------|
+| WASABI_ACCESS_KEY_ID | $developer_wasabi_access_key | dev     |
+| WASABI_SECRET_ACCESS_KEY | $developer_wasabi_secret_access_key | dev     |
+| WASABI_ACCESS_KEY_ID | $developer_wasabi_access_key | staging |
+| WASABI_SECRET_ACCESS_KEY | $developer_wasabi_secret_access_key | staging |
+| WASABI_ACCESS_KEY_ID | $developer_wasabi_access_key | live    |
+| WASABI_SECRET_ACCESS_KEY | $developer_wasabi_secret_access_key | live    |
+| WASABI_DATASETFILES_DIR | wasabi:gigadb-datasets/dev/pub/10.5524 | dev     |
+| WASABI_DATASETFILES_DIR | wasabi:gigadb-datasets/staging/pub/10.5524 | staging |
+| WASABI_DATASETFILES_DIR | wasabi:gigadb-datasets/live/pub/10.5524 | live    |
+| S3_DATASETFILES_DIR| gigadb-datasetfiles:gigadb-datasetfiles-backup/dev/pub/10.5524 | dev     |
+| S3_DATASETFILES_DIR| gigadb-datasetfiles:gigadb-datasetfiles-backup/staging/pub/10.5524 | staging |
+| S3_DATASETFILES_DIR| gigadb-datasetfiles:gigadb-datasetfiles-backup/live/pub/10.5524 | live    |
+
+3. Check that these variables have been added to gitlab `cnhk-infra` variable page
+
+| key | value | env |
+| --- | --- | --- |
+| gigadb_dataset_metadata_aws_access_key_id | xxxxxxxxxxxxxxxxxx | All (default) |
+| gigadb_datasets_metadata_aws_secret_access_key | yyyyyyyyyyyyyyyyy | All (default) |
+
+
+### In dev
+```
+# update the rclone config file
+% cd gigadb/app/tools/transfer-files
```

Comment:
**nitpick**: With all the commands mixed in with command-line output within one single code block, it's easy to miss a step that I need to perform.